### PR TITLE
feat(chatgpt-nvim): add which-key mappings below <leader>G

### PR DESCRIPTION
**Pull request by Kamilcuk**
## 📑 Description

Chatgpt.nvim has mapping recommendation in their readme. I think it makes a nice addition under `<leader>G`, `G` like gpt. I typically use `<space>Gc` to start a chat. Let me know what you think.

image

image

## ℹ Additional Information

### CODE REVIEW

**Review by Uzaaft — APPROVED** [review]

File: lua/astrocommunity/editing-support/chatgpt-nvim/init.lua [hunk 1]
```diff
@@ -5,6 +5,36 @@ return {
...
+              l = {
+                "<cmd>ChatGPTRun code_readability_analysis<CR>",
+                "Code Readability Analysis",
+                mode = { "n", "v" },
+              },
+            },
+          },
+        },
+      },
+    },
```
Comment (Uzaaft) [inline lua/astrocommunity/editing-support/chatgpt-nvim/init.lua:37]:
```suggestion
    },
    { "AstroNvim/astroui", opts = { icons = { ChatGPT = "󰭹" } } },
```

**Comment by Uzaaft** [comment]
Your mappings are wrong.

**Comment by Uzaaft** [comment]
Fixing them myself.

**Comment by Uzaaft** [comment]
@Kamilcuk Can you test it now? :)

File: lua/astrocommunity/editing-support/chatgpt-nvim/init.lua [hunk 2]
```diff
@@ -5,6 +5,53 @@ return {
...
     "nvim-lua/plenary.nvim",
     "nvim-telescope/telescope.nvim",
+    {
+      "AstroNvim/astrocore",
+      opts = function(_, opts)
+        local maps = opts.mappings
+        local prefix = "<Leader>G"
+        maps.n[prefix] = { desc = require("astroui").get_icon("ChatGPT", 1, true) .. "ChatGPT" }
+        maps.v[prefix] = { desc = require("astroui").get_icon("ChatGPT", 1, true) .. "ChatGPT" }
+        maps.n[prefix .. "c"] = { "<cmd>ChatGPT<CR>", desc = "ChatGPT" }
```
Comment (maxbrunet) [inline lua/astrocommunity/editing-support/chatgpt-nvim/init.lua:15]: Could you add `<Leader>GC` for `ChatGPTActAs`, please?

```suggestion
        maps.n[prefix .. "c"] = { "<cmd>ChatGPT<CR>", desc = "ChatGPT" }
        maps.n[prefix .. "c"] = { "<Cmd>ChatGPTActAs<CR>", desc = "ChatGPT Acts As ..." },
```

**Comment by Kamilcuk** [comment]
Hi @Uzaaft . 

> Can you test it now?

I copied the code to my plugins/ and confirm it works nicely.

clarification: So here the code was changed from `dependencies = { "astrocore", opts = { mappings =` into `opts = function(_ , opts) opts.mappigs[...] = ...`. Please kindly satisfy my curiosity. What was the reason for the change?

nonblocking, cosmetics: I am suprised by the choice of 󰭹  as the icon and I prefer and also observe the usage of 🤖 for chatgpt. But I don't care.

Thanks.

**Comment by Uzaaft** [comment]
@maxbrunet How about you, did you test it as well? I think we can combine the changes from your branch into this one if you'd prefer, feel free to suggest those and I'll commit them :) 

@Kamilcuk I'll give you the reason of the change whenever I have time to write an explanation. Might take a while before I have time.

**Review by Uzaaft — APPROVED** [review]

**Pull request merged**